### PR TITLE
Loosen ruby version requirement

### PR DESCRIPTION
**Pull request by mattjarn**
Trying to upgrade an application to ruby 3.0 where we use this as a dependency. Output from rspec run below.

```ruby
ServerSentEvents
  has a version number
  .listen
    produces event from data
  .create_client
    creates new client
    creates new client with headers

ServerSentEvents::Client
  #listen
    produces event from data

ServerSentEvents::Event
  #set
    sets id field
    sets event field
    sets data field
    apends data to existing content
    ignores unknown key 'selection'
    ignores unknown key 'of'
    ignores unknown key 'bad'
    ignores unknown key 'keys'
    ignores unknown key 'here'
  #to_s
    serializes event
    skips missing id
    skips missing event
    serializes empty event

ServerSentEvents::Parser
  #push
    ignores lines starting with :
    emits events on empty line
    parses single event data
    handles multiline data
    emits multiple events
    waits on partial data
    emits all complete events
    parser events with empty values
    parser events with intermixed fields
    strips leading space if present

Finished in 0.03029 seconds (files took 0.3223 seconds to load)
28 examples, 0 failures

Coverage report generated for RSpec to /Users/mj068550/Documents/Git/server-sent-events-ruby/coverage. 76 / 76 LOC (100.0%) covered.
```

### CODE REVIEW

File: lib/server_sent_events/version.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 module ServerSentEvents
-  VERSION = "0.1.2"
+  VERSION = "0.1.3"
```
Comment (tadeboro) [inline lib/server_sent_events/version.rb]: Version number bump has nothing to do with the Ruby version constraint relaxation. This should be in a separate commit or even entirely absent from this PR.

**Review by tadeboro — CHANGES_REQUESTED** [review]

File: lib/server_sent_events/version.rb (hunk 1 above)
Comment (mattjarn) [reply lib/server_sent_events/version.rb]: Gotcha, removed it.

**Review by tadeboro — APPROVED** [review]
Thank you, @mattjarn, for your contribution!

**Comment by tadeboro** [comment]
Changes from this PR are already available in the https://rubygems.org/gems/server_sent_events/versions/0.1.3 Happy eventing!
